src/nvim/ex_getln.c

```diff
@@ -1927,7 +1927,9 @@ static int command_line_changed(CommandLineState *s)
     //       - Immediately undo the effects.
     State |= CMDPREVIEW;
     emsg_silent++;  // Block error reporting as the command may be incomplete
+    msg_silent++;   // Block messages, namely ones that prompt
```

Comment:
Are we sure there are no prompts that are mandatory? You gave the example of "inverse range, are you sure", which should probably be ignored indeed, but is there something else, that might have a bad effect when ignored?

Reply:
That could be the case for more critical errors - I had considered making a new global variable just for the inverse-range prompt, as there's ~100 uses of `msg_silent` in neovim, but I thought this would be too specific a use case.

I've been through the reads of `msg_silent`, and in summary we have:
- Messages:
  - `'shortmess'` display
  - File info display on buffer entry
  - `'showmatch'` when inserting text
  - Search progress / match numbers etc
  - Delay for showing message when tag case mismatch, tag guesses, etc
  - Generic output truncation prompt

- Command output:
  - `'verbose'` >= 15 command line handling
  - Output from `:nmap` and friends
  - `'showmode'`
  - Syntax list output
  - `:messages` history
  - `:silent`

- **Prompts**:
  - Waiting for return on swap file messages
  - Prompting the user for a number (spelling, tag, etc)
  - **Backwards range given**
  - Waiting for return on general commands

- Warnings:
  - File changed since reading (+ associated prompt)
  - Changing a readonly file
  - No write since last change

the rest are just saves, restores or increments/decrements of `msg_silent`. I reckon these are all okay to turn off during substitute interaction, but I'm open to adding a variable just for disabling the prompts.